packages/tldraw/src/lib/shapes/shared/useAsset.ts

```diff
@@ -12,12 +12,13 @@ export function useAsset(shapeId: TLShapeId, assetId: TLAssetId | null, width: n
 	const editor = useEditor()
 	const [url, setUrl] = useState<string | null>(null)
 	const [isPlaceholder, setIsPlaceholder] = useState(false)
+	const isExport = !!useSvgExportContext()
 	const asset = assetId ? editor.getAsset(assetId) : null
 	const culledShapes = editor.getCulledShapes()
 	const isCulled = culledShapes.has(shapeId)
 	const didAlreadyResolve = useRef(false)
 
-	useEffect(() => {
+	useLayoutEffect(() => {
```

Comment:
this was originally to get this resolving _before_ we did an SVG export, but it predates the `useDelaySvgExport` stuff. I've reverted and updated this to use that instead. 

fwiw, i think this would only have caused a double resolve in react strict mode in development